runtime/lua/vim/lsp/completion.lua

```diff
@@ -671,6 +671,153 @@ local function get_augroup(bufnr)
   return string.format('nvim.lsp.completion_%d', bufnr)
 end
 
+---@param winid integer
+---@param bufnr integer
+---@param ft? string
+local function update_popup_window(winid, bufnr, ft)
+  if winid and api.nvim_win_is_valid(winid) and bufnr and api.nvim_buf_is_valid(bufnr) then
+    vim.wo[winid].conceallevel = 2
+    if ft then
+      local lang = vim.treesitter.language.get_lang(ft)
+      if lang then
+        local ok = pcall(vim.treesitter.get_parser, bufnr, lang)
+        if ok then
+          vim.treesitter.start(bufnr, lang)
+        end
+      else
+        vim.wo[winid].eventignorewin = 'OptionSet'
+        vim.bo[bufnr].filetype = ft
+      end
+    end
+    local all = api.nvim_win_text_height(winid, {}).all
+    api.nvim_win_set_height(winid, all)
```

Comment:
I'm experiencing something like an off-by-one with the height, example response:
    `foo:\n```lua\nbar\n```\n`
If I add an extra newline:
    `foo:\n```lua\nbar\n```\n\n`
then `bar` will be shown in the popup.

Reply:
emm I  have fixed this in master.. rebased.